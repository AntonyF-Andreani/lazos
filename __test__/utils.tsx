import React from "react";
import { render } from "@testing-library/react";
import { ThemeProvider } from "@mui/material";

import theme from "theme";

/**
 * Colocar todos los providers, se puede mockear redux y las funciones
 */
const AllTheProviders: React.FC = ({ children }) => {
  return <ThemeProvider theme={theme}>{children}</ThemeProvider>;
};

const customRender = (ui: React.ReactElement) => render(ui, { wrapper: AllTheProviders });

// re-export everything
export * from "@testing-library/react";

// override render method
export { customRender as render };
