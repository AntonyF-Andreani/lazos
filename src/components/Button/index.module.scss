.root {
  border-radius: 35px;
  border: 1px solid transparent;
  color: #fff;
  letter-spacing: 0;
  text-align: center;
  vertical-align: middle;
  opacity: 1;
  text-transform: none;
  padding: 1rem 3rem;
  font-size: 1em;
  font-weight: 700;
  line-height: 1.5;

  &:disabled {
    background: #ededed 0% 0% no-repeat padding-box;
    border-radius: 25px;
    opacity: 1;
  }
}

.contained {
  background-color: #d71920;
  border-color: #d71920 #d71920 #069;

  &:hover {
    background-color: #b6040b;
    border-color: #e6e5e5 #e6e5e5 #ccc;
  }

  &:active,
  &:focus {
    background-color: #b6040b;
    border-color: #e6e5e5 #e6e5e5 #ccc;
    background-image: none;
    box-shadow: none;
  }
}