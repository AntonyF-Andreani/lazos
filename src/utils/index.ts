import { ImageLoaderProps } from "next/image";

/**
 * Utilidad para extraerla URL con formatos optimizados para las imagenes del CMS (Servicio de Imagenes)
 * https://www.storyblok.com/docs/image-service
 */
export const StoryBlockImageURL = ({
  src,
  width,
  quality = 20,
  format = "webp",
  height = 0,
}: StoryBlockImageURLParams) => {
  return `${src}/m/${width}x${height}/filters:quality(${quality}):format(${format})`;
};

interface StoryBlockImageURLParams {
  /** URL de la imagen como la provee Storyblok */
  src: string;
  /** Width of image. The Storyblok API infers aspect ratio */
  width: number;
  /** Quality of image, for default its 20 (20%). */
  quality?: number;
  /** Format of image, for default its webp */
  format?: "webp" | "jpeg" | "png";
  /** Height of the image */
  height?: number;
}

export const StoryBlockLoader = ({ src, width, quality = 20 }: ImageLoaderProps) => {
  return StoryBlockImageURL({ src, width, quality });
};

/**
 * Util for join classes undefined or string and filter undefined
 * @param args classNames
 *
 *  @example const className = joinClasses("a", "b", undefined, "c") // "a b c"
 */
export const joinClasses = (...args: (string | undefined)[]): string => {
  return args.filter((x) => x).join(" "); // undefined is falsey
};

/**
 * Util to test diferrent viewports, must be used with jest.
 *
 * This util make a mock implementation of window.matchMedia
 *
 * @example prepareViewport("(max-width: 768px)")
 */
export const prepareViewport = (mediaQuery: string) => {
  Object.defineProperty(window, "watchMedia", {
    writable: true,
    configurable: true,
    value: mediaQuery,
  });

  window.matchMedia = jest.fn().mockImplementation((query) => {
    return {
      matches: query === mediaQuery,
      media: "",
      onchange: null,
      addListener: jest.fn(),
      removeListener: jest.fn(),
    };
  });
};
