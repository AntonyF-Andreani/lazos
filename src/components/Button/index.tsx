import React, { FC } from "react";
import {
  Button as ButtonMUI,
  Typography,
  TypographyProps,
  ButtonProps as ButtonMUIProps,
} from "@mui/material";

import styles from "./index.module.scss";

import { joinClasses } from "utils";

export interface ButtonProps extends ButtonMUIProps {
  /** Texto del boton */
  text: string;
  /** Propiedades opcionales del componente de texto, por defecto es 'subtitle2' */
  typographyProps?: TypographyProps;
  /** Clase para modificar las clases */
  className?: string;
}

/**
 * Boton primario
 */
const Button: FC<ButtonProps> = ({ text, typographyProps, className, ...anotherProps }) => {
  return (
    <ButtonMUI
      disableElevation
      disableFocusRipple
      classes={{
        root: joinClasses(styles.root, className),
        outlined: styles.outlined,
        contained: styles.contained,
      }}
      disableRipple={anotherProps.variant === "text"}
      {...anotherProps}
    >
      <Typography variant="subtitle2" {...typographyProps}>
        {text}
      </Typography>
    </ButtonMUI>
  );
};

export default Button;
