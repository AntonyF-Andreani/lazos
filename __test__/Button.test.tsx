import React from "react";

import { render } from "./utils";

import Button, { ButtonProps } from "components/Button";

const defaultProps: ButtonProps = {
  text: "test",
};

describe("<Button />", () => {
  it("Should render", () => {
    const { asFragment } = render(<Button {...defaultProps} />);

    expect(asFragment()).toMatchSnapshot();
  });
  it("Should render items", () => {
    const { getByText } = render(<Button {...defaultProps} />);

    expect(getByText("test")).toBeInTheDocument();
  });
});
