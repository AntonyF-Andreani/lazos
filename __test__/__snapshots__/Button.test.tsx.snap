// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button /> Should render 1`] = `
<DocumentFragment>
  <button
    class="MuiButtonBase-root MuiButton-root root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-disableElevation css-s19qo4-MuiButtonBase-root-MuiButton-root"
    tabindex="0"
    type="button"
  >
    <h6
      class="MuiTypography-root MuiTypography-subtitle2 css-7oe211-MuiTypography-root"
    >
      test
    </h6>
    <span
      class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
    />
  </button>
</DocumentFragment>
`;
